src/mapnode.cpp

```diff
@@ -602,13 +602,13 @@ Buffer<u8> MapNode::serializeBulk(int version,
 	// Writing to the buffer linearly is faster
 	u8 *p = &databuf[0];
 	for (u32 i = 0; i < nodecount; i++, p += 2)
-		writeU16(p, nodes[i].param0);
+		writeU16(p, nodes[is_mono_block ? 0 : i].param0);
```

Comment:
so will this (at least branch once, not every node)

Reply:
I can't really do anything else here. Could have a separate implementation of `SerializeBulk` for this case, but that seems overkill.